### PR TITLE
Sorry `IpcCancel_C.thy`

### PR DESCRIPTION
**Pull request by michaelmcinerney**

### CODE REVIEW

File: proof/crefine/RISCV64/SyscallArgs_C.thy [hunk 1]
```diff
@@ -126,11 +124,109 @@ lemma ccorres_pre_getWorkUnits:
...
+lemma isCurDomainExpired_ccorres:
+  "ccorres dc xfdc \<top> UNIV [] isCurDomainExpired (Call isCurDomainExpired_'proc)"
+sorry (* FIXME RT: isCurDomainExpired_ccorres *)
+
 lemma preemptionPoint_ccorres:
   "ccorres (cintr \<currency> dc) (liftxf errstate id (K ()) ret__unsigned_long_')
       invs' UNIV []
       preemptionPoint (Call preemptionPoint_'proc)"
-  apply (cinit simp: workUnitsLimit_def whenE_def)
+  apply (cinit simp: Kernel_Config.workUnitsLimit_def whenE_def) (* FIXME RT: correct use of defn here? *)
```
Comment (lsf37) [inline proof/crefine/RISCV64/SyscallArgs_C.thy:229]: I assume on the C side we're just seeing the plain number, and on the Haskell side we're seeing `Kernel_Config.workUnitsLimit_def`. I think in that case unfolding is legitimate here -- we don't have a nice mechanism yet to make the C side symbolic.

**Review by lsf37 — APPROVED** [review]
Happy with it from my side!

File: proof/crefine/RISCV64/IpcCancel_C.thy [hunk 2]
```diff
@@ -748,7 +762,7 @@ lemma state_relation_queue_update_helper':
    by (auto simp: carch_state_relation_def cmachine_state_relation_def)
 
 lemma state_relation_queue_update_helper:
-  "\<lbrakk> (s, s') \<in> rf_sr; valid_queues s;
+  "\<lbrakk> (s, s') \<in> rf_sr; valid_queues s; \<forall>d p. distinct(ksReadyQueues s (d, p));
```
Comment (corlewis) [inline proof/crefine/RISCV64/IpcCancel_C.thy:765]: nitpick: this condition is missing a space in a couple of other places as well.
```suggestion
  "\<lbrakk> (s, s') \<in> rf_sr; valid_queues s; \<forall>d p. distinct (ksReadyQueues s (d, p));
```

**Review by corlewis — APPROVED** [review]
Nice!

**Pull request merged**